src/main/java/uk/gov/hmcts/reform/feature/service/JwtParserService.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.feature.service;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtParser;
+import uk.gov.hmcts.reform.feature.model.UserTokenDetails;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class JwtParserService {
+
+    private final JwtParser parser;
+
+    public JwtParserService(JwtParser parser) {
+        this.parser = parser;
+    }
+
+    public UserTokenDetails parse(String jwtToken) {
+        try {
+            Claims claims = parser.parseClaimsJws(jwtToken).getBody();
+
+            String id = claims.get("id").toString();
+            List<String> roles = Arrays.asList(claims.get("data").toString().split(","));
+
+            return new UserTokenDetails(id, roles);
+        } catch (Exception exception) {
+            // voiding any exception
+            return null;
```

Comment:
add logging.
Also, shall we continue at this stage, or throw a custom exception?

Reply:
will add.

yes, we should continue no matter what. it is not application nor any other error. always assume as nothing wrong happened. moreover - one should always get a response about feature and filter just provides a way of retrieving user roles